themes/_core/js/product.js

```diff
@@ -166,13 +166,14 @@ function updateProduct(event, eventType, updateUrl) {
         if ($('.images-container').html() !== $newImagesContainer.find('.images-container').html()) {
           $('.images-container').replaceWith(data.product_cover_thumbnails);
         }
-        $('.product-prices').replaceWith(data.product_prices);
-        $('.product-customization').replaceWith(data.product_customization);
-        $('.product-variants').replaceWith(data.product_variants);
-        $('.product-discounts').replaceWith(data.product_discounts);
-        $('.product-additional-info').replaceWith(data.product_additional_info);
-        $('#product-details').replaceWith(data.product_details);
-        $('.product-flags').replaceWith(data.product_flags);
+
+        $('.quickview .product-prices, .page-product:not(.modal-open) .row .product-prices').replaceWith(data.product_prices);
+        $('.quickview .product-customization, .page-product:not(.modal-open) .row .product-customization').replaceWith(data.product_customization);
+        $('.quickview .product-variants .page-product:not(.modal-open) .row .product-variants').replaceWith(data.product_variants);
+        $('.quickview .product-discounts, .page-product:not(.modal-open) .row .product-discounts').replaceWith(data.product_discounts);
+        $('.quickview .product-additional-info, .page-product:not(.modal-open) .row .product-additional-info').replaceWith(data.product_additional_info);
+        $('.quickview #product-details, #product-details').replaceWith(data.product_details);
+        $('.quickview .product-flags, .page-product:not(.modal-open) .row .product-flags').replaceWith(data.product_flags)
```

Comment:
I don't find any element with class `.product-flags` inside the quickview. Consequently, I think this line should only be: `$('.page-product .product-flags').replaceWith(data.product_flags)`.

But this will not fully resolve the problem. A case is when quantity changed from a quickview inside a product page (when product miniatures exist in product page e.g similar products).

Additionnally, quickview doesn't contain: `.product-discounts`, `.product-discounts` and `.product-additionnal-info`.

Let's take an example in detail:

- user is in product X page and there are other similar products Y and Z as miniatures in sidebar.
- user clicks in quickview of Y and changes quantity.

In this DOM, there are :
- 2 elements with class `product-variants` , one in quickview of Y and the other in product page X.
- 2 with `.product-prices`
- 1 element with `.product-customization` but for product X
- 1 element with `#product-details` but for product X
- 1 element with `.product-discounts` but for product X
- 1 element with `.product-additionnal-info` but for product X.


IMHO, the whole problem could be solved by re-naming classes depending on the source (quickview or product page) and keeping javascript as it is for all classes.

Reply:
I agree for the class not contained in quickview, gonna add a commit for it, but for the rest, did you see that I select elements on product X page only if there are no modals opened? It means that product infos will change only if there are no quickview opened

Reply:
Hey, ok this time it looks like you didn't see the .row selector, featured products and product in the same category are not in a .row. I agree with you, thats not really solid, but atleast it avoid breaking changes, we are not able to change JS selectors for a new class because if we would do that, theme which herit these templates or rewrited these but didn't modify the JS would break with a new class, we're able to do it in a major version probably.

Thank you for your feedback and the time you take on this PR :)

Reply:
You pointed me to subject I am not always aware of. Thank you very much for such an invaluable info.

Reply:
No problem, you'r welcome, thanks a lot for your feedback there, it was a pleasure !